### PR TITLE
Update sidebar: new module and fix format

### PR DESCRIPTION
**Pull request by rachelcarmena**
- Add `arrow-core-retrofit`
- Fix format (it solves a problem with the new menu entries because the menu doesn't keep the selected data type cc: @nomisRev)

### CODE REVIEW

**Review by danimontoya — APPROVED** [review]
Thank you @rachelcarmena 🙌

File: docs/_data/sidebar-core.yml [hunk 1]
```diff
@@ -279,6 +279,9 @@ options:
       - title: Core Extensions
         url: /apidocs/arrow-core-data/
 
+      - title: Core Retrofit
+        url: /apidocs/arrow-core-retrofit/
+
       - title: Integrations Retrofit Adapter
         url: /apidocs/arrow-integrations-retrofit-adapter/
```
Comment (aballano) [inline docs/_data/sidebar-core.yml:286]: Do we need to keep this one?

**Review by aballano — APPROVED** [review]

File: docs/_data/sidebar-core.yml (hunk 1 above)
Comment (rachelcarmena) [reply docs/_data/sidebar-core.yml:286]: Thanks for the point @aballano :raised_hands: Yes, that module is still alive with the rest of the content:

```
arrow-integrations-retrofit-adapter/src/main/
└── kotlin
    └── arrow
        └── integrations
            └── retrofit
                └── adapter
                    ├── CallKind2CallAdapter.kt
                    ├── CallKindAdapterFactory.kt
                    ├── CallK.kt
                    ├── ResponseCallback.kt
                    └── syntax.kt
```

 `arrow-core-retrofit` just have part of that module now:

```
arrow-core-retrofit/src/main/
└── kotlin
    └── arrow
        └── retrofit
            └── adapter
                └── either
                    ├── ArrowEitherCallAdapter.kt
                    ├── ArrowResponseECallAdapter.kt
                    ├── eitherAdapter.kt
                    ├── EitherCallAdapterFactory.kt
                    └── ResponseE.kt
```

Note: I didn't show test directories in the trees :wink:

**Comment by rachelcarmena** [comment]
Thanks for the reviews @danimontoya  @aballano :tada:

**Pull request merged**